### PR TITLE
fix(amf): fix for the issue 9151

### LINKED ISSUE #9151: 5G SA | TeraVM Regression | SCTP Abort after Service Request Flow & De Register Req
Test Scenario/issue:
1, SUCI based Registration is successful
2, UE goes to Idle Mode with UE Context Removal flow.
3, Service Request with Signaling initiated by UE & AGW reply with Service Accept
4, UE goes to Idle Mode with UE Context Removal flow.
5, UE sends De Registration Request (Normal) 
6, [Issue]  AGW not sending De Registration Accept, AGW breaks the  SCTP connection with ABORT.

Expectation:
- De registration Accept message after De Register Request ( Normal Dereg)
- SCTP Heartbeat should continue, AGW should not send SCTP ABORT

Tested Build:
1.7.0-1631744238-11c8c453 / 1.7.0-1631759351-95e75e99
Previous Build where this TC passed:
1.7.0-1631565325-eeeb2879

TeraVm 5G SA Test case:
TC7 UE Idle Mode Support (Service Request- Mo Signaling)

Note:
SCTP ABORT in this TC7 is the root cause of TC7a Failures

Logs Enclosed: mme log, syslog & pcap
[SCTP Abort in Idle to Active Mode flow.zip](https://github.com/magma/magma/files/7181318/SCTP.Abort.in.Idle.to.Active.Mode.flow.zip)

Snap of the Issues Traces
Screen Shot 2021-09-16 at 3 27 52 PM

### PR DESCRIPTION
**Pull request by VinayKumar1591**
fix(amf): fix for the issue 9151

## Summary

Added FSM Entry to support DeRegistration in idle mode

## Test Plan

Tested on NGAP Tester

### CODE REVIEW

**Review by ulaskozat — APPROVED** [review]
LGTM

**Comment by ulaskozat** [comment]
can you attach the test result?

File: lte/gateway/c/core/oai/tasks/amf/amf_fsm.cpp [hunk 1]
```diff
@@ -117,6 +117,10 @@ void create_state_matrix() {
       REGISTERED_CONNECTED, STATE_EVENT_DEREGISTER, SESSION_NULL,
       DEREGISTERED_INITIATED, SESSION_NULL, "Deregister_Initiated");
 
+  Update_ue_state_matrix(
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/tasks/amf/amf_fsm.cpp:120]: nit: missed this previously but why is this specific function beginning with uppercase? we should follow the OAI codebase conventions

**Comment by VinayKumar1591** [comment]
please find reference for test results 
[Bug_Fix_issue_9151.zip](https://github.com/magma/magma/files/7187281/Bug_Fix_issue_9151.zip)

**Comment by VinayKumar1591** [comment]
> can you attach the test result?

Done

File: lte/gateway/c/core/oai/tasks/amf/amf_fsm.cpp (hunk 1 above)
Comment (VinayKumar1591) [reply lte/gateway/c/core/oai/tasks/amf/amf_fsm.cpp:120]: somehow from the beginning these functions named like this. other functions in this file also needs to be renamed . Will create Zenhub task for this

**Pull request merged**